pkg/repositories/gormimpl/project_repo.go

```diff
@@ -70,3 +70,33 @@ func NewProjectRepo(db *gorm.DB, errorTransformer errors.ErrorTransformer,
 		metrics:          metrics,
 	}
 }
+
+func (r *ProjectRepo) UpdateProject(ctx context.Context, updatedProject models.Project) (models.Project, error) {
+	var project models.Project
+
+	timer := r.metrics.GetDuration.Start()
+	tx := r.db.Where(&models.Project{
+		Identifier: updatedProject.Identifier,
+	}).First(&project)
+	timer.Stop()
+
+	// Error handling and checking for the result of the database query.
+	if tx.Error != nil {
+		return models.Project{}, r.errorTransformer.ToFlyteAdminError(tx.Error)
+	}
+
+	if tx.RecordNotFound() {
+		return models.Project{}, flyteAdminErrors.NewFlyteAdminErrorf(codes.NotFound, "project [%s] not found", updatedProject.Identifier)
+	}
+
+	// TODO: update the project with the fields and persist to the DB.
+	if updatedProject.Description != "" {
+		project.Description = updatedProject.Description;
+	}
+
+	if len(updatedProject.Labels) > 0 {
+		project.Labels = updatedProject.Labels;
```

Comment:
should we be combining the labels?

Reply:
This is fine. But I have a question on overall update API - it's very tricky one.

In your current implementation, fields cannot be cleared through update. You design is only fields that need to be updated will be passed in ? 

@katrogan Is that model we follow? Or should the recommendation be - do a GET, and change field, and call update. Server does simple overwrite?

Reply:
great point

Reply:
but if we want to do the entity modification, then we will need to use like a update-tag?

Reply:
@schottra advised to use the existing admin.Project structure to construct the request.
> You design is only fields that need to be updated will be passed in.
Correct, if something like ID or Domain is passed on the admin.Project object it will be ignored.
> Or should the recommendation be - do a GET, and change field, and call update. Server does simple overwrite?
I'm not 100% sure what the difference between those two is. In principle, it shouldn't matter if a user of the API knows an ID they want to modify or if they first send a GET request, change the object, then send it back, unless I'm missing something.
In terms of how to actually do the DB write, that's something I'll need some help with since I don't know the API well.

Reply:
i think we use typically use Update (http://gorm.io/docs/update.html#Update-Changed-Fields) which should only update changed fields

Reply:
@kosigz Then I think we would need to make all fields pointers, like say "Description"? 
@katrogan I was referring to the admin API.

Reply:
sorry @anandswaminathan i think the model we're following is using PUT (https://github.com/lyft/flyteidl/pull/78/files#diff-fde1983afefdc036089ed82714330441R320)

i think according to https://github.com/lyft/flyteidl/pull/78#discussion_r469471031 we don't allow for clearing fields in this model

question is - should we? in the sdk we allow for explicitly overriding things like notifications with an empty value